tools/cldr-code/src/main/java/org/unicode/cldr/util/VoteResolver.java

```diff
@@ -1073,6 +1078,29 @@ public int compare(T o1, T o2) {
         }
     };
 
+   /**
+     * Annotate why the O (winning) value is winning vs the N (next) value.
+     * Assumes that the prior annotation mentioned the O value.
+     * @param O optimal value
+     * @param N next-best value
+     */
+    private void annotateNextBestValue(long O, long N, final T oValue, final T nValue) {
+        // See the Comparator<> defined immediately above.
+
+        // sortedValues.size() >= 2 - explain why O won and N lost.
+        // We have to perform the function of the votesThenUcaCollator one more time
+        if (O > N) {
+            annotateTranscript("- This is the winning value because it has the highest weight (voting score).");
+        } else if(winningValue.equals(baselineValue)) {
+            annotateTranscript("- This is the winning value because it is the same as the baseline value, though the weight was otherwise equal to the next-best."); // aka blue star
+        } else if(winningValue.equals(CldrUtility.INHERITANCE_MARKER)) {
+            annotateTranscript("- This is the winning value because it is the inheritance marker, though the weight was otherwise equal to the next-best."); // triple up arrow
+        } else {
+            annotateTranscript("- This is the winning value because it comes earlier than '%s' when the text was sorted, though the weight was otherwise equal to the next-best.", nValue);
+        }
+        annotateTranscript("The Next-best (N) value is '%s', with weight %d", nValue, N);
+    }
```

Comment:
minor quibble: better if subroutine is below its caller

Reply:
in this case however it's intentionally adjacent to votesThenUcaCollator.compare

ideally the annotateTranscript calls might be inside votesThenUcaCollator.compare itself?

Reply:
Right.  
It can't be inside the collator because actually we don't annotate _every_ comparison, just the top two.

Reply:
the collator could call a method that has a boolean to turn on annotation; boolean false when the collator calls it, boolean true when this method calls it just for the top two